### PR TITLE
Better formatting for nested await in member/call expression heads

### PR DESCRIPTION
**Pull request by thorn0**
## Description

Part II for #10340

## Checklist

- [x] I’ve added tests to confirm my change works.
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/common/ast-path.js [hunk 1]
```diff
@@ -146,6 +146,38 @@ class AstPath {
 
     return true;
   }
+
+  /**
+   * Traverses the current node and its parents (heading toward the tree root)
+   * until it finds a node that matches the provided predicate function. Will
+   * return the current node or the matching ancestor. If no such node exists,
+   * it returns undefined.
```
Comment (thorn0) [inline src/common/ast-path.js:154]: Not sure whether including the current node is useful. I did that only to match the behavior of the DOM method. Let's skip the current node and rename the method (suggestions?), shall we?

**Comment by fisker** [comment]
Not very good for short calls 

**Prettier pr-10342**
[Playground link](https://deploy-preview-10342--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuc0DOMAEBDTBeHAd2wEssAKbYszAJzgEcBXODcgSnYDoArNaDgG4AOlEhQMmAEb4ipLADM0XQrTJwAYiQA2ccgp1wANHJqVqWes1YwO3eAA9bnEVBBGQEAA4wS6ZKDYtLQQhAAKQQhoyCDY2sQAntEeUrTYYADWcDAAyl7pJFAA5sgwtCwecAC2UnAAJnX1ADLYxUzYRZoQtFXYML7FMdhMMBDuIAAWMFXaAOoT6mj5YHA5UWQkAG5kCTFgaMkghWhwtDBhaUW9yApxJx58DgBCaZnZOdhVcE2FcDd3xhAjxyhSKugAikwIPB-tp7iB8rQTrQYlJsLVtOMvGpYLMSHUYBNkAAOAAMHmxEBOszSXhi2NYp02fw8zGhcAu3miKGwaAAtFA4PV6uMrEwSPQLh1rkhbnDAScqiRSuUFaCIVCYbKAR4YOi8QSiUgAEy6tI6UEAYQgVRlIFYAFZxkwTgAVdHcuXwzYsACSUEasByYDUPgAggGcjAErpYScAL7xoA)
```sh
--parser babel
```

**Input:**
```jsx
const a = await (await request()).json();
const b = await fs.writeFile(file, await (await request()).text());

```

**Output:**
```jsx
const a = await (
  await request()
).json();
const b = await fs.writeFile(
  file,
  await (
    await request()
  ).text()
);

```

File: src/common/ast-path.js (hunk 1 above)
Comment (fisker) [reply src/common/ast-path.js:154]: `closestAncestor`?

**Review by fisker — APPROVED** [review]

File: src/common/ast-path.js [hunk 2]
```diff
@@ -146,6 +146,37 @@ class AstPath {
...
   }
+
+  /**
+   * Traverses the ancestors of the current node heading toward the tree root
+   * until it finds a node that matches the provided predicate function. Will
+   * return the first matching ancestor. If no such node exists, returns undefined.
+   * @param {(node: any, name: string, number: number | null) => boolean} predicate
+   * @internal Unstable API. Don't use in plugins for now.
+   */
+  findClosestAncestor(predicate) {
```
Comment (fisker) [inline src/common/ast-path.js:157]: Maybe just `findAncestor`

**Review by fisker — APPROVED** [review]

**Pull request merged**